tools/GSDumpGUI/Library/GSDXWrapper.cs

```diff
@@ -205,115 +205,135 @@ public unsafe void Run(GSDump dump, int rendererOverride)
             fixed (byte* pointer = tempregisters)
             {
                 GSsetBaseMem(new IntPtr(pointer));
-                Int32 HWND = 0;
-                GSopen(new IntPtr(&HWND), "", rendererOverride);
+                IntPtr hWnd = IntPtr.Zero;
+
+                if (GSopen(new IntPtr(&hWnd), "", rendererOverride) != 0)
+                    return;
+
                 GSsetGameCRC(dump.CRC, 0);
 
+                NativeMethods.SetClassLong(hWnd,/*GCL_HICON*/ -14, Program.hMainIcon);
+
                 fixed (byte* freeze = dump.StateData)
                 {
-                    byte[] GSFreez = new byte[8];
-                    Array.Copy(BitConverter.GetBytes(dump.StateData.Length), 0, GSFreez, 0, 4);
-                    Array.Copy(BitConverter.GetBytes(new IntPtr(freeze).ToInt32()), 0, GSFreez, 4, 4);
+                    byte[] GSFreez;
+
+                    if (IntPtr.Size > 4)
+                    {
+                        GSFreez = new byte[16];
+                        Array.Copy(BitConverter.GetBytes((Int64)dump.StateData.Length), 0, GSFreez, 0, 8);
+                        Array.Copy(BitConverter.GetBytes(new IntPtr(freeze).ToInt64()), 0, GSFreez, 8, 8);
+                    }
+                    else
+                    {
+                        GSFreez = new byte[8];
+                        Array.Copy(BitConverter.GetBytes((Int32)dump.StateData.Length), 0, GSFreez, 0, 4);
+                        Array.Copy(BitConverter.GetBytes(new IntPtr(freeze).ToInt32()), 0, GSFreez, 4, 4);
+                    }
 
                     fixed (byte* fr = GSFreez)
                     {
                         int ris = GSfreeze(0, new IntPtr(fr));
                         if (ris == -1)
                         {
                             DumpTooOld = true;
-                            return;
+                            Running = false;
                         }
                         GSVSync(1);
 
+                        GSreset();
+                        Marshal.Copy(dump.Registers, 0, new IntPtr(pointer), 8192);
```

Comment:
Magic numbers are dangerous. E.g. I don’t understand where it comes from if you do not even comment it. I guess const int descriptive_name = 8xxx; would be better

Reply:
I have absolutely no clue where that's from, it's old code. I can try to find out and put a comment.

Reply:
No. I thought you added another magic number. I was once considering to add 64bit compatibility without knowing anything about low level programming and basically stopped when I saw all these multiples of 4 going through the code without a single notification.